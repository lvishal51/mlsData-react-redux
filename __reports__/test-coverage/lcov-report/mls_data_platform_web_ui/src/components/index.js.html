<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
