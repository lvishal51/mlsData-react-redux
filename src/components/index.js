export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
