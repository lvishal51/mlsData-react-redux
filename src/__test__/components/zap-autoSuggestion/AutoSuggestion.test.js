import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { AutoSuggestion } from 'components';


const autoSuggestData = [
  {
        "id": 46,
        "mlsInfoId": 2,
        "type": {
            "id": 2,
            "type": "manually completed",
            "parentType": null,
            "lastUpdateTs": "2018-10-04T00:00:00.000Z",
            "createTs": "2018-10-04T00:00:00.000Z"
        },
        "key": "CLASS_OFFICE",
        "value": "CLASS_OFFICE",
        "elasticSearchIndexId": null,
        "hierarchyInfoId": 0,
        "lastUpdateTs": "2018-11-13T00:00:00.000Z",
        "createTs": "2018-11-13T00:00:00.000Z"
    },
    {
        "id": 48,
        "mlsInfoId": 2,
        "type": {
            "id": 2,
            "type": "manually completed",
            "parentType": null,
            "lastUpdateTs": "2018-10-04T00:00:00.000Z",
            "createTs": "2018-10-04T00:00:00.000Z"
        },
        "key": "CLASS_AGENT",
        "value": "CLASS_AGENT",
        "elasticSearchIndexId": null,
        "hierarchyInfoId": 0,
        "lastUpdateTs": "2018-11-13T00:00:00.000Z",
        "createTs": "2018-11-13T00:00:00.000Z"
    }
]

const inputProps = {
  autoSuggestData: autoSuggestData,
  autoSuggestKey: 'key',
  handleInputChange: () => {},
  renderSuggestionItem: () => {},
  onUpdateValueToField: () => {},
  searchKeys: ['key', 'value'],
  suggestionHeader: true,
  renderSuggestionHeader: () => {},
  querySeparator: "=",
  expressionSeparator: ";",
  value: ""
}

const inputPropsWithoutSeparators = {
  autoSuggestData: autoSuggestData,
  autoSuggestKey: 'key',
  handleInputChange: () => {},
  renderSuggestionItem: () => {},
  onUpdateValueToField: () => {},
  searchKeys: ['key', 'value'],
  suggestionHeader: true,
  renderSuggestionHeader: () => {},
  value: ""
}

describe('Zap AutoSuggestion component', () => {
  it('should render correctly.', () => {
    const output = shallow(<AutoSuggestion {...inputProps} />);
    output.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('componentWillUnmount should be called on unmount', () => {
    const component = shallow(<AutoSuggestion {...inputProps} />);
    const componentWillUnmount = jest.spyOn(component.instance(), 'componentWillUnmount');
    component.unmount();
    expect(componentWillUnmount).toHaveBeenCalled();
  });

  it('should render correctly without value prop.', () => {
    const output = shallow(<AutoSuggestion {...inputProps} dataValue=""/>);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'Count' } };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'Count=1', selectionStart: 7 } };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'Count=1;Test=2;', selectionStart: 3 } };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'Count=1;Test=2', selectionStart: 10 } };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, target: { value: ''} };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('onSelectValue function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} value="Class=A" />);
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true, cursorPosition: 7});
    const spy = jest.spyOn(wrapper.instance(), "onSelectValue");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onSelectValue function should be triggered correctly', () => {
    const wrapper = shallow(<AutoSuggestion {...inputPropsWithoutSeparators} />);
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true, cursorPosition: 7});
    const spy = jest.spyOn(wrapper.instance(), "onSelectValue");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onSelectValue function should be triggered correctly when key is selected', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} value='Class=A' />);
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true, cursorPosition: 3 });
    const item = { id: 3 };
    const spy = jest.spyOn(wrapper.instance(), "onSelectValue");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('click', item);
    expect(spy).toBeCalled();
  });

  it('handleInputKeyup function should be triggered correctly when user has selected key and entered =', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 32, ctrlKey: true, target: { selectionStart: 5 } };
    wrapper.setState({textValue: 'Test=', showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleInputKeyup");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('keyUp', event);
    expect(spy).toBeCalled();
  });

  it('handleInputKeyup function should be triggered correctly when user hit ctrl and space at empty field', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 32, ctrlKey: true, target: { selectionStart: 0 } };
    wrapper.setState({textValue: '', showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleInputKeyup");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('keyUp', event);
    expect(spy).toBeCalled();
  });

  it('handleInputKeyup function should be triggered correctly when user press down arrow', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} suggestionHeader={false} />);
    wrapper.setState({textValue: '', autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const event = { preventDefault() {}, which: 40 };
    document.body.innerHTML = '<ListGroupItem class="autoSuggestListItem"></ListGroupItem>';
    const spy = jest.spyOn(wrapper.instance(), "handleInputKeyup");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('keyUp', event);
    expect(spy).toBeCalled();
  });

  it('handleInputKeyup function should be triggered correctly when user presses Semicolon', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, which: 186 };
    const spy = jest.spyOn(wrapper.instance(), "handleInputKeyup");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").at(0).simulate('keyUp', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDownListItem function should be triggered correctly when user press down arrow on list item', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} />);
    const event = { preventDefault() {}, which: 40, target: { nextSibling: {tabindex: 0, focus: jest.fn()}, parentNode: { firstChild: '<div></div>'}}};
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownListItem");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('keyDown', event, 0);
    expect(spy).toBeCalled();
  });

  it('handleKeyDownListItem function should be triggered correctly when user press down arrow on list item', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} suggestionHeader={false} />);
    const event = { preventDefault() {}, which: 40, target: { nextSibling: null, parentNode: { firstChild: {tabindex: 0, focus: jest.fn()}}}};
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownListItem");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('keyDown', event, 0);
    expect(spy).toBeCalled();
  });

  it('handleKeyDownListItem function should be triggered correctly when user press up arrow on list item', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} suggestionHeader={false} />);
    const event = { preventDefault() {}, which: 38, target: { previousSibling: {tabindex: 0, focus: jest.fn()}, parentNode: { lastChild: '<div></div>'}}};
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownListItem");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('keyDown', event, 0);
    expect(spy).toBeCalled();
  });

  it('handleKeyDownListItem function should be triggered correctly when user press up arrow on list item', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} suggestionHeader={false} />);
    const event = { preventDefault() {}, which: 38, target: { previousSibling: null, parentNode: { lastChild: {tabindex: 0, focus: jest.fn()}}}};
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownListItem");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('keyDown', event, 0);
    expect(spy).toBeCalled();
  });

  it('handleKeyDownListItem function should be triggered correctly when user press enter on list item', () => {
    const wrapper = shallow(<AutoSuggestion {...inputProps} / >);
    const event = { preventDefault() {}, which: 13};
    wrapper.setState({autoSuggestData: autoSuggestData, showAutoSuggestionList: true});
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownListItem");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('keyDown', event, 0);
    expect(spy).toBeCalled();
  });
});
