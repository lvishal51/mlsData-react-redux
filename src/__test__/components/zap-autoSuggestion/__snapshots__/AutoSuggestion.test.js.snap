// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap AutoSuggestion component should render correctly without value prop. 1`] = `
<div>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <FormGroup
        className="autoSuggestInput"
        tag="div"
      >
        <Input
          autoComplete="off"
          id="autoComplete"
          innerRef={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
  </Row>
</div>
`;

exports[`Zap AutoSuggestion component should render correctly. 1`] = `
<div>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <FormGroup
        className="autoSuggestInput"
        tag="div"
      >
        <Input
          autoComplete="off"
          id="autoComplete"
          innerRef={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
      <div>
        <ListGroup
          className="autoSuggestList"
          flush={true}
          role="list"
          tag="ul"
        >
          <ListGroupItem
            className="autoSuggestListItem"
            tag="li"
          />
          <ListGroupItem
            className="autoSuggestListItem"
            key="0"
            onClick={[Function]}
            onKeyDown={[Function]}
            tag="li"
          />
          <ListGroupItem
            className="autoSuggestListItem"
            key="1"
            onClick={[Function]}
            onKeyDown={[Function]}
            tag="li"
          />
        </ListGroup>
      </div>
    </Col>
  </Row>
</div>
`;
